docs/extending/add_an_exchange.md

```diff
@@ -132,3 +137,54 @@ Trader.getCapabilities = function () {
   };
 }
 ```
+
+## Error Handling
+
+The various system in Gekko which call exchange functions, ie: the portfolio manager, the paper trader, the importer (if supported) assume that calls to exchange functions will always succeed. Currently, any error returns by an exchange call is treated as an irrecoverable condition and execution of Gekko stops.
+
+In reality calls to web APIs are inherently unreliable, as so it is important to implement a proper error handling and retry system in the exchange to ensure reliable execution and recovery from network conditions.
+
+Refer to the Kraken Exchange implementation for a reference on how to ideally handle retries.
+
+### How to get yourself banned from an exhange
+
+Most exchanges implement ratelimits on their APIs, retrying an API function too quickly or too much is an effective way to have your IP banned from an exchange API. For this reason it is recommended to use a backoff on subsequent retries, and limit retries to a reasonable timeperiod or number of attempts before failing completely.
+
+### Implementing Retry Handling
+
+From `core/util.js` you can schedule a retry by doing the following:
```

Comment:
Maybe instead of posting example code in the doc itself, it might be better to again refer to the kraken implementation. So we don't have to keep this updated if we change anything.